src/mongocxx/collection.cpp

```diff
@@ -65,6 +65,23 @@ stdx::string_view collection::name() const noexcept {
     return _impl->name;
 }
 
+void collection::name(stdx::string_view new_name, bool overwrite_old) {
+  bson_error_t error;
+
+  _impl->name = (std::string)new_name;
```

Comment:
move this line to line 83 in case the rename fails. Also, please just write `_impl->name = new_name`. The cast should not be needed as string_view defines a conversion operator to std::string.